### ISSUE TITLE
Matplotlib issue: ValueError: list.remove(x): x not in list

### ISSUE THREAD

**Issue by dadelforge** [jakobrunge/tigramite#40]
I have just installed the new TIGRAMITE v4 in an anaconda environment running with python 3.7 and matplotlib 3.1.0. While running the tutorials in the Jupyter Notebook environment with the option `%matplotlib inline`. I cannot show the figures obtained for instance with `tigramite.plotting.plot_timeseries`  or `tigramite.plotting.plot_graph`. The functions work well and do return the fig and axes but the fig instance triggers the following error as the jupyter cell runs it. 

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\IPython\core\formatters.py in __call__(self, obj)
    339                 pass
    340             else:
--> 341                 return printer(obj)
    342             # Finally look for special method names
    343             method = get_real_method(obj, self.print_method)

~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\IPython\core\pylabtools.py in <lambda>(fig)
    242 
    243     if 'png' in formats:
--> 244         png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))
    245     if 'retina' in formats or 'png2x' in formats:
    246         png_formatter.for_type(Figure, lambda fig: retina_figure(fig, **kwargs))

~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\IPython\core\pylabtools.py in print_figure(fig, fmt, bbox_inches, **kwargs)
    126 
    127     bytes_io = BytesIO()
--> 128     fig.canvas.print_figure(bytes_io, **kw)
    129     data = bytes_io.getvalue()
    130     if fmt == 'svg':

~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\matplotlib\backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, **kwargs)
   2058                     bbox_artists = kwargs.pop("bbox_extra_artists", None)
   2059                     bbox_inches = self.figure.get_tightbbox(renderer,
-> 2060                             bbox_extra_artists=bbox_artists)
   2061                     pad = kwargs.pop("pad_inches", None)
   2062                     if pad is None:

~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\matplotlib\figure.py in get_tightbbox(self, renderer, bbox_extra_artists)
   2359         bb = []
   2360         if bbox_extra_artists is None:
-> 2361             artists = self.get_default_bbox_extra_artists()
   2362         else:
   2363             artists = bbox_extra_artists

~\AppData\Local\Continuum\anaconda3\envs\tigramite_v4\lib\site-packages\matplotlib\figure.py in get_default_bbox_extra_artists(self)
   2330                 bbox_artists.extend(ax.get_default_bbox_extra_artists())
   2331         # we don't want the figure's patch to influence the bbox calculation
-> 2332         bbox_artists.remove(self.patch)
   2333         return bbox_artists
   2334 

ValueError: list.remove(x): x not in list
```

So far, I can avoid the problem by setting the display mode to `%matplotlib notebook`. I am not sure the problem is related to TIGRAMITE or to my anaconda env but I have not yet encountered this error before while plotting in jupyter notebooks.

**Comment by ewengillies** [comment]
Thanks for the issue report.  While I did refactor the `tigramite` code when moving it to `v4`, I did not look at the plotting functions, so there could definitely be some bugs there.

With that said, I don't see the actual method call from `tigramite` anywhere in the stack trace, so its not easy to tell from a glance where this is from.  Is this the full stack trace?  If so, this tells me that its either:
 * An error due to your specific environment OR
 * there is a bug in `tigramite`

I'm leaning towards option 1 for now given the stack trace.  With that said, lets leave the issue open for a while. If anyone else finds their way to this issue, please comment and we can get to the bottom of it.

**Comment by jakobrunge** [comment]
I tested the tigramite_tutorial_basic.pynb with Python3.7 and Matplotlib 3.0 (which is the default in anaconda for python 3.7) and it works. But I got the same error with matplotlib 3.1.

The error lies in 
`fig, axes = plt.subplots(5, sharex=True, frameon=False, )`
matplotlib3.1 fails with `frameon=False`. 

I would say this is a bug in matplotlib since the frameon parameter is valid. I suggest to use matplotlib 3.0 for now...
